### ISSUE TITLE
Config for 11GB GPU

### ISSUE THREAD

**Issue by justanhduc** [autonomousvision/projected_gan#5]
Hi. This is one of my most anticipated project recently and thanks for finally opensourcing it. It seems like the current training config is made  for 16GB GPUs so I encountered OOM on 11GB ones. Could you please suggest a config that is compatible for smaller GPUs (11GB in my case)? Thanks a lot!

**Comment by justanhduc** [comment]
FYI, I am using a 1024 image database as the default FastGAN also used 1024.

**Comment by xl-sr** [comment]
You have three options:
- train on smaller resolution (duh)
- set e.g. ```--batch-gpu=2``` or the smallest value that still works for you
- reduce the size of the FastGAN generator, e.g., set ngf=64

**Comment by zimonitrome** [comment]
> reduce the size of the FastGAN generator, e.g., set ngf=64

Is it also possible to reduce the number of feature maps of the discriminator?

**Comment by xl-sr** [comment]
you could, but it is not the limiting factor at all. The actual trainable parameters of the discriminator are less than 1M.

**Comment by zimonitrome** [comment]
> you could, but it is not the limiting factor at all. The actual trainable parameters of the discriminator are less than 1M.

Yes, I know. I am actually looking at reducing the discriminator in order to match a much weaker generator, not in order to reduce the memory usage. Just piggy-backing of this issue.

**Comment by xl-sr** [comment]
What happens if you use the PG discriminator with your weaker Generator? Does the discriminator overfit quickly?

**Comment by xl-sr** [comment]
@justanhduc do you still run into memory problems? Actually, the default configuration should work just fine with an 11GB GPU. This is the first tick with the default config:
```
tick 0     kimg 0.1      time 55s          sec/tick 6.3     sec/kimg 98.19   maintenance 48.5   cpumem 3.55   gpumem 6.33   reserved 9.86
```
So, the needed  GPU memory is below 11 GB.

**Comment by zimonitrome** [comment]
@xl-sr in many instances, the discriminator would quickly get a loss of around 0 with the loss of G constantly growing or capping at a high value. Periodically G would briefly outsmart D in which D receives a very high loss for a single iteration until it goes back to 0. Reducing the feature maps by 2 seemed to help, but even better was removing the inner-most projection instead.

I am trying to train an image-to-image GAN so the use-case is not very typical. These generators seem to be much weaker when conditioned on images instead of noise that is sampled or injected akin to the StyleGAN Generator.

**Comment by xl-sr** [comment]
Cool, I want to see how well it works in Im2Im and what the challenges are.

The next thing you should try is setting ```D_kwargs.backbone_kwargs.patch = True```. This might work better you.

**Comment by xl-sr** [comment]
@zimonitrome I am going to close this since it is about memory size originally. Please keep me updated on your progress, via mail or a new issue, I am also interested in the results :)
